AwsCryptographicMaterialProviders/runtimes/java/src/testExamples/java/software/amazon/cryptography/example/hierarchy/concurrent/StorageVersionReadConcurrencyTests.java

```diff
@@ -0,0 +1,302 @@
+        }
+      }
+    )
+    .build();
+
+  @BeforeClass
+  public void setup() {
+    dateFormat.setTimeZone(timeZone);
+    threadIndexToStorage = new ConcurrentHashMap<>(16, 1, threadCount);
+    threadIndexToKeyStore = new ConcurrentHashMap<>(16, 1, threadCount);
+    storageIndexToThreadId = new ConcurrentHashMap<>(16, 1, threadCount);
+    keyStoreIndexToThreadId = new ConcurrentHashMap<>(16, 1, threadCount);
+    versionKeyOutputMap = new ConcurrentHashMap<>(16, 1, threadCount);
+
+    unpickedIndicesForStorage = new ConcurrentLinkedDeque<>(identifiers);
+    unpickedIndicesForKeyStore = new ConcurrentLinkedDeque<>(identifiers);
+    // For every identifier which will ultimately map to one thread, we will create a unique
+    // storage layer per thread with a unique ddb client. This will make it so that
+    // we isolate resources even further and prevent resource reuse.
+    identifiers.forEach(id -> {
+      threadIndexToStorage.put(id, createStorageLayer());
+      threadIndexToKeyStore.put(id, createKeyStore());
+    });
+
+    encryptionContext = new HashMap<>();
+    encryptionContext.put("custom", "ec");
+
+    final DynamoDbClient _ddbClient = DynamoDbClient.create();
+    DynamoDBTable table = DynamoDBTable
+      .builder()
+      .ddbClient(_ddbClient)
+      .ddbTableName(Fixtures.TEST_KEYSTORE_NAME)
+      .build();
+    Storage tmp = Storage.builder().ddb(table).build();
+    KeyStoreAdmin admin = KeyStoreAdmin
+      .builder()
+      .KeyStoreAdminConfig(
+        KeyStoreAdminConfig
+          .builder()
+          .storage(tmp)
+          .logicalKeyStoreName(Fixtures.TEST_KEYSTORE_NAME)
+          .build()
+      )
+      .build();
+    CreateKeyInput createKeyInput = CreateKeyInput
+      .builder()
+      .Identifier(branchKeyId)
+      .EncryptionContext(encryptionContext)
+      .KmsArn(
+        KmsSymmetricKeyArn
+          .builder()
+          .KmsKeyArn(Fixtures.KEYSTORE_KMS_ARN)
+          .build()
+      )
+      .build();
+    admin.CreateKey(createKeyInput);
+    System.out.println(
+      "Successfully set up test with branch key: " + branchKeyId
+    );
+  }
+
+  @AfterClass
+  public void teardown() {
+    final DynamoDbClient _ddbClient = DynamoDbClient.create();
+    DdbHelper.deleteKeyStoreDdbItem(
+      branchKeyId,
+      "branch:ACTIVE",
+      Fixtures.TEST_KEYSTORE_NAME,
+      DynamoDbClient.create(),
+      true
+    );
+    DdbHelper.deleteKeyStoreDdbItem(
+      branchKeyId,
+      "beacon:ACTIVE",
+      Fixtures.TEST_KEYSTORE_NAME,
+      DynamoDbClient.create(),
+      true
+    );
+    QueryResponse res = _ddbClient.query(queryRequestForCleanUp);
+    res
+      .items()
+      .forEach(response -> {
+        DdbHelper.deleteKeyStoreDdbItem(
+          branchKeyId,
+          response.get("type").s(),
+          Fixtures.TEST_KEYSTORE_NAME,
+          DynamoDbClient.create(),
+          true
+        );
+      });
```

Comment:
Blocking: Use `software.amazon.cryptography.example.DdbHelper#DeleteBranchKey` to delete all the items of a BK.

Reply:
I tried using it but I kept finding items that belonged to the branch key id i used, but maybe i used it wrong

Reply:
ahh I used a different method `DeleteBranchKey` works